xbmc/network/GUIDialogNetworkSetup.cpp

```diff
@@ -191,20 +200,23 @@ void CGUIDialogNetworkSetup::InitializeSettings()
   // Add our protocols
   TranslatableIntegerSettingOptions labels;
 #ifdef HAS_FILESYSTEM_SMB
-  labels.push_back(std::make_pair(20171, NET_PROTOCOL_SMB));
+  m_protocols.emplace_back(Protocol{true, true, true, false, true, 0, "smb", 20171});
 #endif
-  labels.push_back(std::make_pair(20301, NET_PROTOCOL_HTTPS));
-  labels.push_back(std::make_pair(20300, NET_PROTOCOL_HTTP));
-  labels.push_back(std::make_pair(20254, NET_PROTOCOL_DAVS));
-  labels.push_back(std::make_pair(20253, NET_PROTOCOL_DAV));
-  labels.push_back(std::make_pair(20173, NET_PROTOCOL_FTP));
-  labels.push_back(std::make_pair(20175, NET_PROTOCOL_UPNP));
-  labels.push_back(std::make_pair(20304, NET_PROTOCOL_RSS));
+  const std::vector<Protocol> defaults =
+        {{ true,  true,  true,  true, false, 443, "https", 20301},
+         { true,  true,  true,  true, false,  80,  "http", 20300},
+         { true,  true,  true,  true, false, 443,  "davs", 20254},
+         { true,  true,  true,  true, false,  80,   "dav", 20253},
+         { true,  true,  true,  true, false,  21,   "ftp", 20173},
+         {false, false, false, false,  true,   0,  "upnp", 20175},
```

Comment:
missing space

Reply:
do not agree. trailing space before } either.